src/rendervulkan.cpp

```diff
@@ -264,8 +268,58 @@ bool CVulkanTexture::BInit( uint32_t width, uint32_t height, VkFormat format, cr
 	{
 		assert( format == DRMFormatToVulkan( pDMA->format ) );
 	}
+	fprintf(stderr, "m: %d %p %lx %x\n", g_vulkanSupportsModifiers, pDMA, pDMA ? pDMA->modifier : 0, usage);
+	if ( g_vulkanSupportsModifiers && pDMA && pDMA->modifier != DRM_FORMAT_MOD_INVALID )
+	{
+		VkPhysicalDeviceImageDrmFormatModifierInfoEXT modifierFormatInfo = {};
+		VkPhysicalDeviceExternalImageFormatInfo externalImageFormatInfo = {};
+		VkPhysicalDeviceImageFormatInfo2 imageFormatInfo = {};
+		VkExternalImageFormatProperties externalImageProperties = {};
+		VkImageFormatProperties2 imageProperties = {};
+		
+		modifierFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_DRM_FORMAT_MODIFIER_INFO_EXT;
+		modifierFormatInfo.drmFormatModifier = pDMA->modifier;
+		modifierFormatInfo.sharingMode = imageInfo.sharingMode;
+
+		externalImageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_IMAGE_FORMAT_INFO;
+		externalImageFormatInfo.pNext = &modifierFormatInfo;
+		externalImageFormatInfo.handleType = VK_EXTERNAL_MEMORY_HANDLE_TYPE_DMA_BUF_BIT_EXT;
+
+		imageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2;
+		imageFormatInfo.pNext = &externalImageFormatInfo;
+		imageFormatInfo.format = imageInfo.format;
+		imageFormatInfo.type = imageInfo.imageType;
+		imageFormatInfo.tiling = VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT;
+		imageFormatInfo.usage = imageInfo.usage;
+		imageFormatInfo.flags = imageInfo.flags;
+		
+		externalImageProperties.sType = VK_STRUCTURE_TYPE_EXTERNAL_IMAGE_FORMAT_PROPERTIES;
+
+		imageProperties.sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2;
+		imageProperties.pNext = &externalImageProperties;
+		
+		if ( vkGetPhysicalDeviceImageFormatProperties2(physicalDevice, &imageFormatInfo, &imageProperties) == VK_SUCCESS &&
```

Comment:
We should probably enumerate supported formats/modifiers on startup, so that we can intersect them with e.g. KMS formats/modifiers. But this can be left for later.

Reply:
Given that at this point this imports just with the provided modifier (or at least tries to if allowed) I think that isn't needed yet.

(The main problem with enumeration is that wlroots currently exposes the GL supported modifiers instead of the Vulkan supported modifiers)

Reply:
>The main problem with enumeration is that wlroots currently exposes the GL supported modifiers instead of the Vulkan supported modifiers

Yup! Working on a fix for this.

Reply:
I've pushed my work in https://github.com/Plagman/gamescope/pull/160.